### PR TITLE
Upgrade to use tasks_v2 library

### LINKED ISSUE #1: Support new Google tasks python library
Upgrade dependencies to use the new library. 
https://googleapis.dev/python/cloudtasks/latest/index.html

### PR DESCRIPTION
**Pull request by JuliaGuimiot**
Closes #1

### CODE REVIEW

File: django_cloud_tasks/connection.py [hunk 1]
```diff
@@ -48,9 +51,12 @@ def __get__(self, obj, cls):
 class GoogleCloudClient(object):
     @cached_property
     def client(self):
-        client = googleapiclient.discovery.build('cloudtasks', 'v2beta3',
-                credentials=DCTConfig.google_cloud_credentials(), 
-                cache=DiscoveryCache())
+        client = googleapiclient.discovery.build(
```
Comment (sww314) [inline django_cloud_tasks/connection.py]: you want to replace this client with the new version.

File: django_cloud_tasks/connection.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import googleapiclient.discovery
-import os.path
 import hashlib
+import os.path
 import tempfile
 
+import googleapiclient.discovery
```
Comment (sww314) [inline django_cloud_tasks/connection.py]: this goes away, replace with new library

File: django_cloud_tasks/connection.py [hunk 3]
```diff
@@ -12,14 +13,16 @@ class DiscoveryCache:
 
     See https://github.com/googleapis/google-api-python-client/issues/325#issuecomment-419387788
     """
+
```
Comment (sww314) [inline django_cloud_tasks/connection.py]: this whole class (DiscoverCache) can go away

File: django_cloud_tasks/base.py [hunk 4]
```diff
@@ -161,27 +171,31 @@ def __init__(self, body):
         self.setup()
 
     def setup(self):
-        payload = self.body['task']['appEngineHttpRequest']['body']
+        payload = self.body["task"]["appEngineHttpRequest"]["body"]
```
Comment (sww314) [inline django_cloud_tasks/base.py]: this is appengine specific, I think our task `payload` has to look different.

**Review by sww314 — COMMENTED** [review]
I made some notes. I suggest we run black on the entire code base and merge that change to master. 
It will make it easier to deal with these diffs.

File: django_cloud_tasks/base.py (hunk 4 above)
Comment (JuliaGuimiot) [reply django_cloud_tasks/base.py]: fixed

File: django_cloud_tasks/connection.py (hunk 3 above)
Comment (JuliaGuimiot) [reply django_cloud_tasks/connection.py]: fixed

File: django_cloud_tasks/connection.py (hunk 2 above)
Comment (JuliaGuimiot) [reply django_cloud_tasks/connection.py]: fixed

File: django_cloud_tasks/connection.py (hunk 1 above)
Comment (JuliaGuimiot) [reply django_cloud_tasks/connection.py]: fixed

**Review by sww314 — APPROVED** [review]

**Pull request merged**